### ISSUE TITLE
Update atlas version for JEDI applications

### ISSUE THREAD

**Issue by srherbener** [JCSDA/spack-stack#702]
**Please describe the package or library you would like to add to spack-stack.**

Need to update atlas version per requirements for the upcoming atlas interface changes in oops. The specific version of atlas is to be determined.

- [x] atlas version 0.34.0

**What applications will be using this package or library?**

JEDI applications including interpolations, VADAR and SABER

**Is there already a package or library in spack-stack that provides this, or related, functionality?**

Atlas is already a part of spack-stack, just need to update to a newer version

**Additional context**

None

**Will This Package be Needed in a NOAA Operational Application?**
If yes, the WCOSS System Software Checklist must also be completed.

**WCOSS System Software Request Checklist**

General questions:

Supervisor or sponsor of the requester

Software name and version, specific URL to the software

Software type - New or Upgrade existing?

Justification ( List NPS models using the software)

Completion Time requested

Software License including all Dependency Software Licenses 

Support contact(s) who must have a WCOSS account

Dependency Software list

Installation instructions

Test and verification instructions

Technical Review Checklist for open source software - Review the source code to answer the following questions

Licenses for the requested software and its dependencies

Licenses -  confirm the software Licenses are acceptable

Maturity

Acceptable - Stable, production, or equivalent

Self-contained

No external http, https,  ftp, or other URI exists except that in comments

No binary files in the package unless they are in the approved list

No publicly disclosed cybersecurity vulnerabilities and exposures 

Searching https://cve.mitre.org/cve/

Security High Level Checklists 

Is it prohibited by DHS/DOC/NOAA/NWS

Is it provided by a trusted source? Trusted sources include other NWS, NOAA, or DOC, agencies, or other Federal agencies that operate at a FISMA high or equivalent level. Additionally, trusted sources could be third-party agencies through which there is an existing SLA on file (such as RedHat). 

Is software support offered (is it being updated and patched). 

If maintained by a private entity, does the entity operate in a foreign country, especially a prohibited foreign country (China, Russia, Iran, North Korea, etc.). 

Is there sufficient documentation to support maintenance 

Are there known vulnerabilities or weaknesses

Is there a need for privileged processes 

Are there software dependencies, are those dependencies approved or do they have any security concerns 

Are there any other concerns related to SA, SI, and SC control families

**Comment by fmahebert** [comment]
@ytremolet and I think the next spack-stack release should include the most recent atlas: tag `0.34.0`.

However, we think the atlas module should _not_ be included in the meta-modules by default. This is because we'll be developing and merging JEDI features that require develop atlas in the next months, and needing to unload a module that's included by default risks being confusing and error-prone. We'd rather put atlas into the bundle and give folks the option of custom loading an atlas module if it makes more sense for their needs.

Does this make sense?

**Comment by climbfuji** [comment]
> @ytremolet and I think the next spack-stack release should include the most recent atlas: tag `0.34.0`.
> 
> However, we think the atlas module should _not_ be included in the meta-modules by default. This is because we'll be developing and merging JEDI features that require develop atlas in the next months, and needing to unload a module that's included by default risks being confusing and error-prone. We'd rather put atlas into the bundle and give folks the option of custom loading an atlas module if it makes more sense for their needs.
> 
> Does this make sense?

Yeaaaaaaa ... somewhat. I understand the need to do this for the upcoming release, but it's moving as away from the goal to eventually release JEDI as spack packages. I won't make the change on spack-stack develop, that one will get tag 0.34.0 and have it as default in jedi-base-env. But on the release branch for spack-stack-1.5.0, I'll take atlas out of jedi-base-env (which prevents loading the module by default) but still build the tag you mentioned. Would that be ok?

**Comment by fmahebert** [comment]
Sounds ok to me, but I'd like @ytremolet to give an opinion as well.

**Comment by ytremolet** [comment]
> Sounds ok to me, but I'd like @ytremolet to give an opinion as well.

Yes, that's fine.

**Comment by climbfuji** [comment]
As agreed upon, we'll do this on the spack-stack 1.5.0 release branches only, not on develop.

**Comment by climbfuji** [comment]
Ok, after some back and forth and considering the implications of not building and loading the `ecmwf-atlas` module in `skylab-dev`/`unified-dev`, we decided to not do this. We build `ecmwf-atlas` and load it as part of the JEDI modules as before. A tiny guard in `jedi-bundle` will stop the `cmake` process if the `ecmwf-atlas` module is loaded, and tell the users to unload it.

Since the version has already been updated to the latest available version (0.34.0), this issue can be closed.
